stan/math/prim/meta/as_column_vector_or_scalar.hpp

```diff
@@ -16,7 +16,7 @@ namespace math {
  * @param a Specified scalar.
  * @return 1x1 matrix that contains the value of scalar.
  */
-template <typename T>
+template <typename T, typename = require_stan_scalar_t<T>>
```

Comment:
[comment]

Recently I've been trying to use more 'nice' names for template types. Like for this the template type could be Scalar, functions that take in matrices can have a template type of `Mat`. Not needed for this PR but for future reference I've found it to make my code a bit nicer to read

Reply:
While I agree, I think this should go in a separate PR as this one is big enough.

Reply:
^yes agree